tests/src/deque/src/main.sw

```diff
@@ -0,0 +1,37 @@
+script;
+
+use deque::Deque;
+use std::assert::assert;
+
+fn main() -> bool {
+    let mut deque_push_front = Deque::new();
+    deque_push_front.push_front(1);
+    deque_push_front.push_front(2);
+    deque_push_front.push_front(3);
+
+    assert(deque_push_front.pop_front().unwrap() == 3);
```

Comment:
There are additional combinations for `push_front()`, `push_back()`, `pop_front()` & `pop_back` that should be tested here because popping affects the `front` and `back` fields depending on how the struct has been set up for the test.